twitter-eventlistener-plugin/src/main/java/com/twitter/presto/plugin/eventlistener/QueryCompletedEventScriber.java

```diff
@@ -93,6 +116,13 @@ private static QueryCompletionEvent toThriftQueryCompletionEvent(QueryCompletedE
       thriftEvent.distributed_planning_time_ms = eventStat.getDistributedPlanningTime().get().toMillis();
     }
     thriftEvent.total_bytes = eventStat.getTotalBytes();
+    if (eventMetadata.getPayload().isPresent()) {
+      JsonReader jsonReader = Json.createReader(new StringReader(eventMetadata.getPayload().get()));
+      long totalDataTransferBytes = getTotalDataTransferInBytes(jsonReader.readObject());
+      // uncomment after new thrift def is built and uploaded
+      //thriftEvent.total_data_transfer_bytes = totalDataTransferBytes;
```

Comment:
We verified the input data size is the same as the output data size of the previous stage. So to avoid double counting, and excluding the very first input data size(fetch from HDFS), I think summing up all output data size is the correct formula for data transfer.
Agree with your previous comment though that it's better to leave things at the more granular level.